pyfstat/helper_functions.py

```diff
@@ -327,19 +327,43 @@ def convert_array_to_gsl_matrix(array):
     return gsl_matrix
 
 
-def get_sft_array(sftfilepattern, data_duration, F0, dF0):
-    """ Return the raw data from a set of sfts """
+def get_sft_array(sftfilepattern, data_duration=None, F0=None, dF0=None):
+    """ Return the raw data (absolute value) from a set of SFTs
+
+    FIXME: currently only returns data for first detector
+    """
+
+    if F0 is None and dF0 is None:
+        fMin = -1
+        fMax = -1
+    elif F0 is None or dF0 is None:
+        raise ValueError("Need either none or both of F0, dF0.")
+    else:
+        fMin = F0 - dF0
+        fMax = F0 + dF0
+
+    if data_duration is not None:
+        logging.warning(
+            "Option 'data_duration' for get_sft_array()"
+            " is no longer in use and will be removed."
+        )
 
     SFTCatalog = lalpulsar.SFTdataFind(sftfilepattern, lalpulsar.SFTConstraints())
-    MultiSFTs = lalpulsar.LoadMultiSFTs(SFTCatalog, F0 - dF0, F0 + dF0)
+    MultiSFTs = lalpulsar.LoadMultiSFTs(SFTCatalog, fMin, fMax)
+    ndet = MultiSFTs.length
+    if ndet > 1:
+        logging.warning(
+            "Loaded SFTs from {:d} detectors, only using the first.".format(ndet)
+        )
+
     SFTs = MultiSFTs.data[0]
-    data = []
-    for sft in SFTs.data:
-        data.append(np.abs(sft.data.data))
+    times = [sft.epoch.gpsSeconds for sft in SFTs.data]
+    data = [np.abs(sft.data.data) for sft in SFTs.data]
     data = np.array(data).T
-    n, nsfts = data.shape
-    freqs = np.linspace(sft.f0, sft.f0 + n * sft.deltaF, n)
-    times = np.linspace(0, data_duration, nsfts)
+    nbins, nsfts = data.shape
+
+    sft0 = SFTs.data[0]
+    freqs = np.linspace(sft0.f0, sft0.f0 + nbins * sft0.deltaF, nbins)
```

Comment:
@dbkeitel Maybe you already did the math, but shouldn't this be
```
freqs = np.linspace(sft0.f0, sft0.f0 + (nbins-1) * sft0.deltaF, nbins)
```
since `np.linspace` *includes* the very last point? Alternatively 
```
freqs = np.linspace(sft0.f0, sft0.f0 + nbins * sft0.deltaF, nbins, endpoint=False)
```
Examples:
```
>>> import numpy as np
>>> np.linspace(0, 10, 10)
array([ 0.        ,  1.11111111,  2.22222222,  3.33333333,  4.44444444,
        5.55555556,  6.66666667,  7.77777778,  8.88888889, 10.        ])
>>> np.linspace(0, 9, 10)
array([0., 1., 2., 3., 4., 5., 6., 7., 8., 9.])
>>> np.linspace(0, 10, 10, endpoint=False)
array([0., 1., 2., 3., 4., 5., 6., 7., 8., 9.])
```

Reply:
Good catch! This was already present in the original code, so I did in fact not check the math. Will fix.

Reply:
done in https://github.com/PyFstat/PyFstat/pull/125/commits/99a6a90e785292be3a610aed59f8db09e2f71b04